pvlib/iotools/solcast.py

```diff
@@ -0,0 +1,322 @@
+    latitude, longitude, api_key, map_variables=True, **kwargs
+):
+    """Get the irradiance and weather for a Typical Meteorological Year (TMY) at a requested location.
+
+    Derived from satellite (clouds and irradiance over non-polar continental areas) and
+    numerical weather models (other data). The TMY is calculated with data from 2007 to 2023.
+
+    Parameters
+    ----------
+    latitude : float
+        in decimal degrees, between -90 and 90, north is positive
+    longitude : float
+        in decimal degrees, between -180 and 180, east is positive
+    api_key : str
+        To access Solcast data you will need an API key: https://toolkit.solcast.com.au/register.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    kwargs:
+        Optional parameters passed to the API. See https://docs.solcast.com.au/ for full list of parameters.
+
+    Returns
+    -------
+    df : pandas.DataFrame
+        containing the values for the parameters requested
+
+    Examples
+    --------
+    get_solcast_tmy(
+        latitude=-33.856784,
+        longitude=151.215297,
+        output_parameters='dni,ghi',
+        api_key="your-key"
+    )
+    """
+
+    params = dict(
+        latitude=latitude,
+        longitude=longitude,
+        format="json",
+        **kwargs
+    )
+
+    return _get_solcast(
+        endpoint="tmy/radiation_and_weather",
+        params=params,
+        api_key=api_key,
+        map_variables=map_variables
+    )
+
+
+def get_solcast_historic(
+    latitude,
+    longitude,
+    start,
+    api_key,
+    end=None,
+    duration=None,
+    map_variables=True,
+    **kwargs
+):
+    """Get historical irradiance and weather estimated actuals
+
+    for up to 31 days of data at a time for a requested location,
+    derived from satellite (clouds and irradiance
+    over non-polar continental areas) and numerical weather models (other data).
+    Data is available from 2007-01-01T00:00Z up to real time estimated actuals.
+
+    Parameters
+    ----------
+    latitude : float
+        in decimal degrees, between -90 and 90, north is positive
+    longitude : float
+        in decimal degrees, between -180 and 180, east is positive
+    start : datetime-like
+        First day of the requested period
+    end : optional, datetime-like
+        Last day of the requested period
+    duration : optional, default is None
+        Must include one of end_date and duration. ISO_8601 compliant duration for the historic data.
+        Must be within 31 days of the start_date.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    api_key : str
+        To access Solcast data you will need an API key: https://toolkit.solcast.com.au/register.
+    kwargs:

```

Comment:
to consiter:

- add surface_tilt, surface_azimuth in the pvlib convetion, remap them to tilt/azimuth query params
- add all current query parameters to the function signature - rather than saying "you have to check the solcast docs". I know this is a different philosophy to the sdk, but I think maybe worth it for this use case

Reply:
decision for time being is "ask in solcast format".